generate_sample.go

```diff
@@ -304,6 +305,42 @@ func newKafkaSample() *exoscalev1.Kafka {
 	}
 }
 
+func generateOpensearchSample() {
+	spec := newOpensearchSample()
+	serialize(spec, true)
+}
+
+func newOpensearchSample() *exoscalev1.OpenSearch {
+	return &exoscalev1.OpenSearch{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       exoscalev1.OpenSearchKind,
+			APIVersion: exoscalev1.OpenSearchGroupVersionKind.GroupVersion().String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "opensearch-local-dev",
+		},
+		Spec: exoscalev1.OpenSearchSpec{
+			ResourceSpec: xpv1.ResourceSpec{},
```

Comment:
```suggestion
			ResourceSpec: xpv1.ResourceSpec{
				ProviderConfigReference:          &xpv1.Reference{Name: "provider-config"},
				WriteConnectionSecretToReference: &xpv1.SecretReference{Name: "opensearch-local-dev-details", Namespace: "default"},
			},
```

You need to specify the provider config otherwise it points to a non existing config `default` and then it fails.